File: Java/Easy/Accepted/second-largest-digit-in-a-string_7.java
class Solution {
    public int secondHighest(String s) {
        
        int[] count = new int[10];
        for (char c : s.toCharArray()) {
            if (Character.isDigit(c)) {
                count[c - '0']++;
            }
        }
        int max = -1;
        int secondMax = -1;
        for (int i = 0; i < 10; i++) {
            if (count[i] > 0) {
                if (i > max) {
                    secondMax = max;
                    max = i;
                } else if (i > secondMax) {
                    secondMax = i;
                }
            }
        }
        return secondMax;
    }
}

File: C++/Easy/Accepted/second-largest-digit-in-a-string_7.cpp
class Solution {
public:
    int secondHighest(string s) {
        
        int largest = -1;
        int second = -1;
        for (auto c : s) {
            if (c >= '0' && c <= '9') {
                int n = c - '0';
                if (n > largest) {
                    second = largest;
                    largest = n;
                } else if (n > second && n != largest) {
                    second = n;
                }
            }
        }
        return second;
    }
};